operator/v1/types_vsphere_problem_detector.go

```diff
@@ -0,0 +1,66 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VSphereProblemDetector is used to manage and report health of Openshift
+// cluster running in vSphere environment.
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VSphereProblemDetector objects allows configuration and health reporting of
+// Vsphere problem detector operator.
+type VSphereProblemDetector struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec VSphereProblemDetectorSpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status VSphereProblemDetectorStatus `json:"status"`
+}
+
+// VSphereProblemDetectorSpec is the desired configuration of vSphere problem detector operator.
+type VSphereProblemDetectorSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+// VSphereHealthCheckStatus is the status of vSphere problem detector operator and
+// health of Openshift and vSphere integration.
+type VSphereProblemDetectorStatus struct {
+	// List of checks and their status
+	Checks         []VSphereHealthCheck `json:"vSphereHealthCheck,omitempty"`
+	OperatorStatus `json:",inline"`
+}
+
+type VSphereHealthCheck struct {
+	Name          string             `json:"name"`
+	Result        VSphereCheckResult `json:"vSphereCheckResult"`
+	Message       string             `json:"message"`
+	LastCheckTime metav1.Time        `json:"lastCheckTime,omitempty"`
+}
+
+type VSphereCheckResult string
+
+const (
+	VSphereCheckPassed  VSphereCheckResult = "passed"
+	VSphereCheckFailed  VSphereCheckResult = "failed"
+	VSphereCheckWarning VSphereCheckResult = "warning"
```

Comment:
usually constants are upper-case

Reply:
converted these into vars. They read weird if all upper case.

Reply:
API constants are upper case ("passed" -> "Passed", as we have"True" and not "true" in conditions)

Reply:
fixed.